dogu-actions/prepare/src/main.ts

```diff
@@ -34,10 +24,38 @@ ActionKit.run(async ({ options, logger, input, deviceHostClient, consoleActionCl
   logger.info('resolve checkout path... from', { DOGU_ROUTINE_WORKSPACE_PATH, checkoutPath });
   const resolvedCheckoutPath = path.resolve(DOGU_ROUTINE_WORKSPACE_PATH, checkoutPath);
   logger.info('resolved checkout path', { resolvedCheckoutPath });
-
   await checkoutProject(logger, consoleActionClient, deviceHostClient, resolvedCheckoutPath, clean, branch, tag, checkoutUrl);
 
-  const appPath = await downloadApp(logger, consoleActionClient, deviceHostClient, DOGU_DEVICE_PLATFORM, DOGU_HOST_WORKSPACE_PATH, currentPlatformAppVersion);
+  let appPath = '';
+  const resolvedAppVersion = appVersion || process.env.DOGU_APP_VERSION || '';
+  const resolvedAppPackageName = appPackageName || process.env.DOGU_APP_PACKAGE_NAME || '';
+
+  if (resolvedAppPackageName) {
+    const currentPlatformPackageName =
+      typeof resolvedAppPackageName === 'object'
+        ? (() => {
+            const platformAppVersion = Reflect.get(resolvedAppPackageName, DOGU_DEVICE_PLATFORM) as string | undefined;
+            if (!platformAppVersion) {
+              throw new Error(`Invalid app version: ${stringify(resolvedAppPackageName)} for platform: ${DOGU_DEVICE_PLATFORM}`);
```

Comment:
Is the error message incorrect?. I think it should be Invalid app package name: ...